app.js

```diff
@@ -399,3 +399,25 @@ form.addEventListener('submit', (event) => {
     error.className = 'error';
   }
 });
+
+/* Preserve to local storage */
+
+if (localStorage.getItem('formData')) {
+  const formData = JSON.parse(localStorage.getItem('formData'));
+  document.getElementById('fullname').value = formData.name;
+  document.getElementById('mail').value = formData.email;
+  document.getElementById('message').value = formData.message;
+}
+
+document.getElementById('contact-form').addEventListener('submit', (event) => {
+  event.preventDefault();
+  const formData = {
+    name: document.getElementById('fullname').value,
+    email: document.getElementById('mail').value,
+    message: document.getElementById('message').value,
+  };
+
+  localStorage.setItem('formData', JSON.stringify(formData));
+
+  document.getElementById('contact-form').reset();
```

Comment:
- [ ] Kindly consider adding `window.onload` function that checks if there's any saved data in the local storage and if so the input fields are pre-filled with the data. you can do so below 
- hint :
```
 window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```